lib/gala/payment_token.rb

```diff
@@ -20,14 +21,15 @@ def initialize(token_attrs)
       self.transaction_id = headers["transactionId"]
       self.ephemeral_public_key = headers["ephemeralPublicKey"]
       self.public_key_hash = headers["publicKeyHash"]
+      self.application_data = headers["applicationData"]
     end
 
     def decrypt(certificate_pem, private_key_pem)
+      self.class.validate_signature(signature, ephemeral_public_key, data, transaction_id, application_data)
```

Comment:
FWIW, most doc refers to the validation of the signature or the tag (depending on the encryption scheme) as part of the decryption process.  For example, https://en.wikipedia.org/wiki/Integrated_Encryption_Scheme  lists it as decryption step 3.